packages/clerk-js/src/ui/components/UserProfile/ConnectedAccountsSection.tsx

```diff
@@ -105,9 +104,16 @@ const ConnectedAccountAccordion = ({ account }: { account: ExternalAccountResour
               return user
                 .createExternalAccount({
                   strategy: account.verification!.strategy as OAuthStrategy,
-                  redirect_url: isModal
-                    ? appendModalState({ url: window.location.href, componentName })
-                    : window.location.href,
+                  redirect_url: appendRedirectState({
```

Comment:
@desiprisg this looks kinda like spaghetti. Can we extract it somehow to look cleaner?

Reply:
I plan on extracting this after adding a `routing` prop to the router context.